builtin/credential/approle/path_login_test.go

```diff
@@ -301,3 +301,56 @@ func generateRenewRequest(s logical.Storage, auth *logical.Auth) *logical.Reques
 
 	return renewReq
 }
+
+func TestAppRole_RoleResolve(t *testing.T) {
+	var resp *logical.Response
+	var err error
+	b, storage := createBackendWithStorage(t)
+
+	role := "role1"
+	createRole(t, b, storage, role, "a,b,c")
+	roleRoleIDReq := &logical.Request{
+		Operation: logical.ReadOperation,
+		Path:      "role/role1/role-id",
+		Storage:   storage,
+	}
+	resp, err = b.HandleRequest(context.Background(), roleRoleIDReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("err:%v resp:%#v", err, resp)
+	}
+	roleID := resp.Data["role_id"]
+
+	roleSecretIDReq := &logical.Request{
+		Operation: logical.UpdateOperation,
+		Path:      "role/role1/secret-id",
+		Storage:   storage,
+	}
+	resp, err = b.HandleRequest(context.Background(), roleSecretIDReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("err:%v resp:%#v", err, resp)
+	}
+	secretID := resp.Data["secret_id"]
+
+	loginData := map[string]interface{}{
+		"role_id":   roleID,
+		"secret_id": secretID,
```

Comment:
It doesn't look like the approle `ResolveRoleOperation` handler requires a `secret_id`. Is this necessary?

Reply:
It isn't necessary, but it's what would be passed to it in reality, as this is what a valid login request would look like (and we will use the arguments to the login request to resolve the role). I'm inclined to leave it in, as it's meant to work when given the same data as a valid login request.

Reply:
Makes sense, thanks for the context.